manifest/list.go

```diff
@@ -33,6 +33,10 @@ type List interface {
 	// must be specified.
 	UpdateInstances([]ListUpdate) error
 
+	// RemoveInstance removes the instance with the given digest from the list. If no such instance
+	// exists the list is left unchanged.
+	RemoveInstance(digest.Digest)
```

Comment:
This should probably return an error on a missing digest (but because it start as an internal-only interface, it’s not critical to get right).

Reply:
I'm afraid my knowledge of go and this project make it difficult for me to perform such a refactoring. So instead I moved the functionality to a private method using a type switch. That way it has no impact on the public interface and can it safely be refactored in a future PR.

Reply:
> This should probably return an error on a missing digest (but because it will start as an internal-only interface, it’s not critical to get right).

I personally prefer the more idempotent type of behavior (why return an error if the end result is what you want it to be), but I'm fine of course to implement it differently. Since this is internal now indeed I left it as is.